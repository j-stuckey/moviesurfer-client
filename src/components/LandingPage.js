import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Redirect, Link } from 'react-router-dom';

class LandingPage extends Component {
    render() {
        const { isLoggedIn } = this.props;

        return (
            <div>
                {isLoggedIn && <Redirect to="/search" />}
                <h1>Welcome to Moviesurfer!</h1>
                <p>Welcome page still in development...</p>

                <Link to="/register">Sign Up</Link>
				<Link to="/login">Login</Link>
            </div>
        );
    }
}

const mapStateToProps = state => ({
    isLoggedIn: state.auth.currentUser !== null
});

export default connect(mapStateToProps)(LandingPage);
