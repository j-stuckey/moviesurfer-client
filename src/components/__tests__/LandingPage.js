import React from 'react';
import { LandingPage } from '../LandingPage';
import { shallow } from 'enzyme';

describe('<LandingPage />', () => {
    const wrapper = shallow(<LandingPage />);

    it('renders without crashing', () => {
        expect(wrapper.exists()).toBe(true);
    });

    it('renders correctly', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
