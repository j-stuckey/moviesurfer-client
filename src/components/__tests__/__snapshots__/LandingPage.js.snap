// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LandingPage /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LandingPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        <h1>
          Welcome to Moviesurfer!
</h1>,
        <p>
          Welcome page still in development...
</p>,
        <Link
          replace={false}
          to="/register"
>
          Sign Up
</Link>,
        <Link
          replace={false}
          to="/login"
>
          Login
</Link>,
      ],
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Welcome to Moviesurfer!",
        },
        "ref": null,
        "rendered": "Welcome to Moviesurfer!",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Welcome page still in development...",
        },
        "ref": null,
        "rendered": "Welcome page still in development...",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Sign Up",
          "replace": false,
          "to": "/register",
        },
        "ref": null,
        "rendered": "Sign Up",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Login",
          "replace": false,
          "to": "/login",
        },
        "ref": null,
        "rendered": "Login",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          <h1>
            Welcome to Moviesurfer!
</h1>,
          <p>
            Welcome page still in development...
</p>,
          <Link
            replace={false}
            to="/register"
>
            Sign Up
</Link>,
          <Link
            replace={false}
            to="/login"
>
            Login
</Link>,
        ],
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome to Moviesurfer!",
          },
          "ref": null,
          "rendered": "Welcome to Moviesurfer!",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome page still in development...",
          },
          "ref": null,
          "rendered": "Welcome page still in development...",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Sign Up",
            "replace": false,
            "to": "/register",
          },
          "ref": null,
          "rendered": "Sign Up",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Login",
            "replace": false,
            "to": "/login",
          },
          "ref": null,
          "rendered": "Login",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
